src/diagnostics.rs

```diff
@@ -171,3 +206,15 @@ pub enum DiagnosticError {
         serde_json::Error,
     ),
 }
+
+pub trait ErrorDiagnostic {
+    fn diagnostic(&self) -> String;
+}
+
+#[cfg(feature = "diagnostics")]
```

Comment:
Is this necessary? This module is already guarded by a `cfg` on the `diagnostics` feature.

Reply:
Double protection! (Good catch)